### PR TITLE
Tiny refactor of sample caching service worker

### PR DESCRIPTION
**Pull request by mohawk2**
Stumbled across this code from https://developer.mozilla.org/en-US/docs/Web/API/Cache

Comments/suggested changes welcome!

### CODE REVIEW

**Comment by mohawk2** [comment]
I signed it!

File: service-worker/selective-caching/service-worker.js [hunk 1]
```diff
@@ -29,20 +29,27 @@ var CURRENT_CACHES = {
   font: 'font-cache-v' + CACHE_VERSION
 };
 
+function hashValues(hash) {
+  return Object.keys(hash).map(function(key) { return hash[key]; });
```
Comment (ebidel) [inline service-worker/selective-caching/service-worker.js]: no need for this function.  `Object.values(CURRENT_CACHES)` gives you the object's values.

File: service-worker/selective-caching/service-worker.js [hunk 2]
```diff
@@ -29,20 +29,27 @@ var CURRENT_CACHES = {
...
+
 self.addEventListener('activate', function(event) {
   // Delete all caches that aren't named in CURRENT_CACHES.
   // While there is only one cache in this example, the same logic will handle the case where
   // there are multiple versioned caches.
-  var expectedCacheNames = Object.keys(CURRENT_CACHES).map(function(key) {
-    return CURRENT_CACHES[key];
-  });
-
+  var expectedCacheNamesSet = array2set(hashValues(CURRENT_CACHES));
```
Comment (ebidel) [inline service-worker/selective-caching/service-worker.js]: I think we should just create a proper `new Set()` and use it here. `array2set` is uneeded

**Review by ebidel — CHANGES_REQUESTED** [review]

File: service-worker/selective-caching/service-worker.js (hunk 1 above)
Comment (mohawk2) [reply service-worker/selective-caching/service-worker.js]: Learn something new every day, thanks! Done.

File: service-worker/selective-caching/service-worker.js (hunk 2 above)
Comment (mohawk2) [reply service-worker/selective-caching/service-worker.js]: Also quite right! Done.

**Comment by mohawk2** [comment]
I made the suggested changes. Was wondering what the process was for any next steps?

**Comment by beaufortfrancois** [comment]
If @ebidel is OK, we'll merge it.

**Comment by mohawk2** [comment]
@ebidel Any objections?

**Comment by mohawk2** [comment]
...

**Comment by mohawk2** [comment]
@beaufortfrancois Been over a year now?

**Comment by beaufortfrancois** [comment]
@ebidel WDYT?

**Comment by mohawk2** [comment]
...

**Comment by mohawk2** [comment]
Bueller?

**Comment by mohawk2** [comment]
Bueller?

**Comment by mohawk2** [comment]
Another year? Merge this, please.

**Comment by mohawk2** [comment]
@beaufortfrancois Been another year.

**Comment by ebidel** [comment]
@jeffposnick should look at this.

**Comment by jeffposnick** [comment]
Thanks for the heads-up.

**Comment by mohawk2** [comment]
Hallelujah!

**Pull request merged**